### DISCUSSION TITLE
Configurable image media formats (support for HEIC)

### DISCUSSION THREAD

**Discussion by odinns** [spatie/laravel-medialibrary#2257] [Ideas] [answered]
Hi guys and thanks for this library!

I am a proud personal sponsor of Spatie 👍 

I see that supported image formats are simple arrays in `Conversions/ImageGenerators/Image.php`.

Would you be open to make that array configurable via `config/media-library.php`?

I would be happy to do a PR for you, to expedite a merge of such a feature, rather than having to pre-convert before adding to medialibrary.

Specifically I am asking because we want to support the HEIF/HEIC format, and do conversions of it to other formats directly with the medialibrary. It seems that the only obstacle is that the supported formats are hardcoded. 

I already have the servers configured with support in ImageMagick, so the only missing bit is this.

Let me know what you think 😊

**Comment by freekmurze** [comment, marked as answer]
Sure! Send a PR for us to review 👍 

Thank you for sponsoring us ❤️
